### PR TITLE
Turn a class annotated with @Startup but no scope into a bean

### LINKED ISSUE #7048: Allow the standalone use of @Startup
Currently the new `@Startup` annotation requires a scope annotation to go along with it.

It was [suggested](https://github.com/quarkusio/quarkus/pull/6969#issuecomment-582784981) to make `@Startup` a standalone annotation that is `@ApplicationScoped` by default

### PR DESCRIPTION
**Pull request by mkouba**
- add @ApplicationScope automatically
- resolves #7048

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Weird, I tried this and it didn't work for me. I must have made some subtle mistake :)

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java [hunk 1]
```diff
@@ -45,6 +49,31 @@
...
+        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+
+            @Override
+            public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
+                return kind == org.jboss.jandex.AnnotationTarget.Kind.CLASS;
+            }
+
+            @Override
+            public void transform(TransformationContext context) {
+                if (context.isClass() && !BuiltinScope.isDeclaredOn(context.getTarget().asClass())) {
```
Comment (manovotn) [inline extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java]: I know it's somewhat far-fetched but should we care about non-built in scopes? Users can have custom ones, or there is `Transactional` one already present. Both cases will pass this check and try to add yet another scope on the bean.

**Review by manovotn — COMMENTED** [review]
Ah, I see you beat me to this PR. Was planning on sending it but got distracted by other issues...
I left one comment, otherwise it looks good as there is no black magic in this PR :)

File: extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java (hunk 1 above)
Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java]: That's a good point. Although in this case I doubt that a `@Startup` bean would be annotated with `@Transactional` or a custom scope. Let's create a separate issue to handle this corretly. WDYT?

Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java]: Hm, the problem is that annotation transformers are registered before the final set of scopes is known...

Comment (mkouba) [reply extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java]: I've created https://github.com/quarkusio/quarkus/issues/7076

**Pull request merged**